src/components/Hoverable/index.js

```diff
@@ -1,189 +1,198 @@
-        if (prevProps.disabled === this.props.disabled) {
-            return;
-        }
+            if (shouldHandleScroll && isScrolling.current) {
+                return;
+            }
+            setIsHovered(hovered);
+        },
+        [disabled, shouldHandleScroll],
+    );
 
-        if (this.props.disabled && this.state.isHovered) {
-            this.setState({isHovered: false});
-        }
-    }
+    useEffect(() => {
+        const unsetHoveredWhenDocumentIsHidden = () => document.visibilityState === 'hidden' && setIsHovered(false);
 
-    componentWillUnmount() {
-        document.removeEventListener('visibilitychange', this.handleVisibilityChange);
-        document.removeEventListener('mouseover', this.checkHover);
-        if (this.scrollingListener) {
-            this.scrollingListener.remove();
-        }
-    }
+        document.addEventListener('visibilitychange', unsetHoveredWhenDocumentIsHidden);
 
-    /**
-     * Sets the hover state of this component to true and execute the onHoverIn callback.
-     *
-     * @param {Boolean} isHovered - Whether or not this component is hovered.
-     */
-    setIsHovered(isHovered) {
-        if (this.props.disabled) {
-            return;
-        }
+        return () => document.removeEventListener('visibilitychange', unsetHoveredWhenDocumentIsHidden);
+    }, []);
 
-        /**
-         * Capture whther or not the user is hovering over the component.
-         * We will use this to determine if we should update the hover state when the user has stopped scrolling.
-         */
-        this.isHoveredRef = isHovered;
-
-        /**
-         * If the isScrollingRef is true, then the user is scrolling and we should not update the hover state.
-         */
-        if (this.isScrollingRef && this.props.shouldHandleScroll && !this.state.isHovered) {
+    useEffect(() => {
+        if (!shouldHandleScroll) {
             return;
         }
 
-        if (isHovered !== this.state.isHovered) {
-            this.setState({isHovered}, isHovered ? this.props.onHoverIn : this.props.onHoverOut);
-        }
-    }
+        const scrollingListener = DeviceEventEmitter.addListener(CONST.EVENTS.SCROLLING, (scrolling) => {
+            isScrolling.current = scrolling;
+            if (!scrolling) {
+                setIsHovered(isHoveredRef.current);
+            }
+        });
+
+        return () => scrollingListener.remove();
+    }, [shouldHandleScroll]);
 
     /**
      * Checks the hover state of a component and updates it based on the event target.
      * This is necessary to handle cases where the hover state might get stuck due to an unreliable mouseleave trigger,
      * such as when an element is removed before the mouseleave event is triggered.
      * @param {Event} e - The hover event object.
      */
-    checkHover(e) {
-        if (!this.wrapperView || !this.state.isHovered) {
+    const unsetHoveredIfOutside = useCallback(
+        (e) => {
```

Comment:
Let's rename to use normal names.
```suggestion
        (element) => {
```

Reply:
https://github.com/Expensify/App/pull/29052#discussion_r1350071462